test/test.py

```diff
@@ -0,0 +1,39 @@
+import subprocess
```

Comment:
This functionality does not seem to justify the use of Python. A bash script would be able to handle this fine.

Reply:
i just used what was easier for me

Reply:
Ok, personal preference is not a good justification for tool choices. Optimize for readability and ease of use for other contributors, not yourself.

Shell is the default choice because it's portable and very easy to execute CLIs in a series of commands. If there were something especially difficult to handle in shell (concurrency, parsing, math, etc.), that's a good reason to introduce another tool.

Reply:
I don't think our team has decided on a default choice.

We use Python in a few GCP, Azure, and Linux host monitoring areas. It also seems like more members of our team are familiar with Python than Shell.

Shell is portable, but portability isn't the primary feature in this case. I think dev velocity is more important for now, so I leaned towards a tool that is easy for me to use but also doesn't raise to barrier to entry for others.

Reply:
I'm flagging this not as a Python vs. Shell thing. At some point shell gets unwieldy and a proper language with better type support. I'd generally prefer to see Go, more because of well-established tools like https://pkg.go.dev/github.com/gruntwork-io/terratest than a fundamental language difference, but that's not the issue here.

The meat of this program, lines 34-43, is a shell program. The Python bits around it handle flag parsing and a loop. Both things that shell does well. If you were parsing the output of `output -json` into a dictionary, doing complex assertions, etc., I wouldn't ask you to shoehorn that into a shell script. But right now you're doing all the hard parts in a shell, and Python just ends up being a distraction for negligible benefit, if not net cost.

Consider, for example, how signals are handled. If you `SIGINT` a test run, does it terminate early and gracefully? In a shell, you can just change `terraform` to `exec terraform` and Terraform becomes the foreground process and receives signals directly. In an interpreted program which spawns a shell, you have multiple layers through which signals need to be propagated. Possible for sure, but shell has its strengths, and spawning CLIs in series and printing their output is shell's ultimate strength.

Reply:
i changed it to shell, seems to have been easier than i thought, thanks.